core/src/cats/postgresql.cc

```diff
@@ -828,6 +844,7 @@ void BareosDbPostgresql::SqlUpdateField(int i)
 
 SQL_FIELD* BareosDbPostgresql::SqlFetchField(void)
 {
+  AssertOwnership();
```

Comment:
do we have any idea how this will influence runtime?
`SqlFetchField()` is called a lot in pretty tight loops.
Its happy-path boils down to:
1. integer comparison
2. nullptr check
3. boolean check
4. add two pointers (for array access)
5. integer increment

Calling `AssertOwnership()` while assuming we're on Linux, it is completly inlined, on the happy path, and having `is_private_ == false`:
1. boolean check
2. pointer dereference
3. integer equality comparison
4. integer comparison
5. library call to `pthread_self()` (cannot be inlined)
7. long integer equality comparison (always inlined from `pthread_equal()`)

I don't say this will be a performance issue. But it definitely looks  like could be. Especially with the redundant check in `SqlUpdateField()` above we would call `AssertOwnership()` twice for each field.

Reply:
What do you think of this:  We leave the calls where they are, but use an empty definition during release builds so that we only do the checking on debug builds ?